### PR TITLE
MARP-2631 Restructure end2end unit test

### PR DESCRIPTION
**Pull request by tvtphuc-axonivy**

### CODE REVIEW

**Review by ntqdinh-axonivy — CHANGES_REQUESTED** [review]

File: unit-tester/src/main/java/com/axonivy/ivy/unittester/constants/UnitTesterConstants.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package com.axonivy.ivy.unittester.constants;
+
+public class UnitTesterConstants {
```
Comment (nqhoan-axonivy) [inline unit-tester/src/main/java/com/axonivy/ivy/unittester/constants/UnitTesterConstants.java:3]: It should be End2EndConstants, because I found that only related to end2end

**Comment by tvtphuc-axonivy** [comment]
Hi @ivy-rew , currently in the test folder of the connectors, there are a lot of duplicated codes. We would like to move common codes into this repository, so we just need to call methods from this repository for the connectors.
Here is the PR I made as an example: https://github.com/axonivy-market/skribble-connector/pull/23/files.

**Comment by ivy-lli** [comment]
I don't think that it makes much sense that you move code from your control under our control and share such common code over different repos.
What if you or we change this here, then you will break 20 repo builds?
I think every repo needs to know how to set up such variables etc. and it makes no sense to over-engineer this.

**Comment by ivy-rew** [comment]
> Hi @ivy-rew , currently in the test folder of the connectors, there are a lot of duplicated codes. We would like to move common codes into this repository, so we just need to call methods from this repository for the connectors. Here is the PR I made as an example: https://github.com/axonivy-market/skribble-connector/pull/23/files.

If you need to share this code, I'd invent an own utility which is under your jurisdiction. But at is is no way attached to what we do here in web-tester, unit-tester in the general market universe I don't think this is the right spot here.

However, I have one more concern why I would not deploy such test-template pattern over repo boundaries. IMHO it's a nice approach which you found there and it may works well in many situations.
But for many cases we could setup the same thing in a much simpler way. E.g. by inventing a TestName suffix that will automatically be execute as E2e test. While others are CI tests.
The name convention could be something like "MyCool**E2eTest**". IMHO this would simply the Market infra a lot. It provides more flexibility as not all mock tests necessarily have to be e2e tests. Still you could internally use the Test-Pattern approach where it works good for you.
